webview-ui/src/components/AnalysisPage/AnalysisPage.tsx

```diff
@@ -90,7 +90,7 @@ const AnalysisPage: React.FC = () => {
     if (enhancedIncidents.length > 0 && serverRunning) {
       dispatch(getSuccessRate());
     }
-  }, [enhancedIncidents.length, serverRunning, dispatch]);
+  }, [enhancedIncidents, serverRunning, dispatch]);
```

Comment:
Using **enhancedIncidents** runs the effect too often because the array reference keeps changing, even if the content doesn't. Results in too many requests and lead to a memory leak warnings. May be triggering this only when array length changes is a better option.